src/Uno.UI/Microsoft/UI/Xaml/Controls/NavigationView/NavigationView.xaml

```diff
@@ -10,136 +10,200 @@
+								<VisualState x:Name="ClosedCompact">
+									<VisualState.Setters>
+										<Setter Target="PaneAutoSuggestBoxPresenter.Visibility" Value="Collapsed"/>
+										<Setter Target="PaneAutoSuggestButton.Visibility" Value="Visible"/>
+									</VisualState.Setters>
+								</VisualState>
+							</VisualStateGroup>
+
+							<VisualStateGroup x:Name="PaneStateListSizeGroup">
+								<VisualState x:Name="ListSizeFull"/>
+								<VisualState x:Name="ListSizeCompact">
+									<VisualState.Setters>
+										<Setter Target="PaneContentGrid.Width" Value="{Binding RelativeSource={RelativeSource TemplatedParent}, Path=CompactPaneLength}"/>
+										<Setter Target="PaneTitleTextBlock.Visibility" Value="Collapsed" />
+										<Setter Target="PaneHeaderContentBorder.Visibility" Value="Collapsed" />
+										<Setter Target="PaneCustomContentBorder.HorizontalAlignment" Value="Left"/>
+										<Setter Target="FooterContentBorder.HorizontalAlignment" Value="Left"/>
+									</VisualState.Setters>
+								</VisualState>
+							</VisualStateGroup>
+
+							<VisualStateGroup x:Name="TitleBarVisibilityGroup">
+								<VisualState x:Name="TitleBarVisible" />
+								<VisualState x:Name="TitleBarCollapsed">
+									<VisualState.Setters>
+										<Setter Target="PaneContentGrid.Margin" Value="0,32,0,0"/>
+									</VisualState.Setters>
+								</VisualState>
+							</VisualStateGroup>
+
+							<VisualStateGroup x:Name="OverflowLabelGroup">
+								<VisualState x:Name="OverflowButtonWithLabel" />
+								<VisualState x:Name="OverflowButtonNoLabel">
+									<VisualState.Setters>
+										<Setter Target="TopNavOverflowButton.Style" Value="{ThemeResource NavigationViewOverflowButtonNoLabelStyleWhenPaneOnTop}" />
+									</VisualState.Setters>
+								</VisualState>
+							</VisualStateGroup>
+
+							<VisualStateGroup x:Name="BackButtonGroup">
+								<VisualState x:Name="BackButtonVisible" />
+								<VisualState x:Name="BackButtonCollapsed">
+									<VisualState.Setters>
+										<Setter Target="BackButtonPlaceholderOnTopNav.Width" Value="0" />
+									</VisualState.Setters>
+								</VisualState>
+							</VisualStateGroup>
+
+							<VisualStateGroup x:Name="PaneVisibilityGroup">
+								<VisualState x:Name="PaneVisible" />
+								<VisualState x:Name="PaneCollapsed">
+									<VisualState.Setters>
+										<!-- Note that RootSplitView.DisplayMode is set in code so we don't want to -->
+										<!-- write it here and interfere. But these values work together to hide -->
+										<!-- the left pane. -->
+										<Setter Target="RootSplitView.CompactPaneLength" Value="0" />
+										<Setter Target="PaneToggleButtonGrid.Visibility" Value="Collapsed" />
+									</VisualState.Setters>
+								</VisualState>
+							</VisualStateGroup>
+						</VisualStateManager.VisualStateGroups>
+
+						<!-- Button grid -->
+						<Grid
+                            x:Name="PaneToggleButtonGrid"
```

Comment:
See the 'Uno specific' comment on `PaneToggleButtonGrid` in the existing XAML - presumably it should stick with the workaround, since ZIndex is still not supported on Grid (https://github.com/unoplatform/uno/issues/325)